### PR TITLE
Refactoring: Move platform specific code files to src/

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**

Moves the platform specific code to src/

CHANGELOG: Refactoring: Moved around files to clean up the repository
**Context: Fixes an issue?**

Related to: https://github.com/jamulussoftware/jamulus/pull/2601

**Does this change need documentation? What needs to be documented and how?**

No. 

**Status of this Pull Request**

Ready for testing and review.

**What is missing until this pull request can be merged?**
Testing.. Should either compile or not compile to show if everything is working.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [ ] I've filled all the content above

### CODE REVIEW

File: Jamulus.pro [hunk 1]
```diff
@@ -246,8 +246,8 @@ win32 {
 
     HEADERS += src/sound/oboe/sound.h
 
-    SOURCES += src/sound/oboe/sound.cpp \
-        android/androiddebug.cpp
+SOURCES += src/sound/oboe/sound.cpp \
```
Comment (pljones) [inline Jamulus.pro:249]: Indent?

Comment (ann0see) [reply Jamulus.pro]: Fixed.

**Review by pljones — APPROVED** [review]
On the basis that a clean build _for all targets affected_ works...

**Comment by ann0see** [comment]
I'd wait until the CI is green and then test the artifacts.

**Review by pljones — APPROVED** [review]

**Review by hoffie — APPROVED** [review]
Yay, thanks. :)

**Review by ann0see — COMMENTED** [review]
@hoffie @pljones can one of you please test Android?

**Comment by ann0see** [comment]
macOS legacy build works as expected and shows the badge. I can NOT run the normal build, as something seems to be wrong with the universal build. It doesn't show up as universal but ARM now?

**Comment by ann0see** [comment]
iOS also installs and works correctly.

**Comment by pljones** [comment]
> @hoffie @pljones can one of you please test Android?

Downloaded the ZIP, downloaded an unzipper (eventually... why doesn't Android have one?!)... uninstalled Jamulus (as I always forget this wasn't immediate...)...  Installed build from ZIP successfully.

**Comment by ann0see** [comment]
Ok. Great. So I'll merge it.

**Pull request merged**